### PR TITLE
ProgressBar Graph Integration

### PR DESCRIPTION
**Pull request by nxanil**
@natabal 

Convert `ProgressBar Graph` into the functional component using various HOCs and new library `material-ui/core`. Please have a look.

### CODE REVIEW

File: GraphsV2/ProgressBarGraph/index.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        width,
+        height,
+        properties
+    } = props;
+
+    useEffect(() => {
+        setTooltip();
+    }, [props.data, props.width, props.height])
+
+    const setTooltip = () => {
```
Comment (natabal) [inline GraphsV2/ProgressBarGraph/index.js]: you don't really need this function, let's remove and add the code below in useEffect.

File: GraphsV2/ProgressBarGraph/index.js [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+            fontSize={fontSize}
+            width={availableWidth}
+            height={height}
+        >
+            {
+                data.map((barData, i) => {
+                    const customTooltip = !isEmpty(customTooltips) ? customTooltips.tooltipProps(barData) : {};
+
+                    return (
+                        <Item
```
Comment (natabal) [inline GraphsV2/ProgressBarGraph/index.js]: let's add an id for identification

**Comment by nxanil** [comment]
`ProgressBar Graph`

progress

**Comment by natabal** [comment]
@nxanil Can we place the graph in the middle of the visualization? And we should by default have a smaller width for each bar.

File: GraphsV2/ProgressBarGraph/index.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        colorRange,
+        units,
+        maxDataFormat,
+        defaultRange,
+        fontSize,
+        maxSectionHeight,
+        id
+    } = properties;
+
+    const availableWidth = width - (margin.left + margin.right);
```
Comment (natabal) [inline GraphsV2/ProgressBarGraph/index.js]: I am thinking that dimension calculations should be done only once not on every re-render. Let's calculate once on mounting the component and then save and re-use.

**Pull request merged**